includes/wc-core-functions.php

```diff
@@ -205,7 +210,12 @@ function wc_get_template_part( $slug, $name = '' ) {
  * @param string $default_path  Default path. (default: '').
  */
 function wc_get_template( $template_name, $args = array(), $template_path = '', $default_path = '' ) {
-	$cache_key = sanitize_key( implode( '-', array( 'template', $template_name, $template_path, $default_path, WC_VERSION ) ) );
+	if ( isset( $_SERVER['SERVER_NAME'] ) ) {
+		$uniquepath = sanitize_key( $_SERVER['SERVER_NAME'] );
+	} else {
+		$uniquepath = '';
+	}
+	$cache_key = sanitize_key( implode( '-', array( 'template', $template_name, $template_path, $default_path, WC_VERSION, $uniquepath ) ) );
```

Comment:
Here we can do the same as I suggested before, but here the filter should be `woocommerce_get_template_cache_key`.

Reply:
Thanks for the feedback @claudiosanches , updated my PR here https://github.com/woocommerce/woocommerce/pull/23783/commits/217a95e937b6a2b5e54b0e5ee1023c4ae6cc7946 as per your advice